packages/react-components/src/components/chart/hooks/useDataZoom.ts

```diff
@@ -0,0 +1,64 @@
+import { MutableRefObject, useEffect, useState } from 'react';
+import { useViewport } from '../../../hooks/useViewport';
+import { EChartsType } from 'echarts';
+import { ECHARTS_GESTURE } from '../../../common/constants';
+import { DEFAULT_DATA_ZOOM } from '../eChartsConstants';
+import { ViewportInMs } from '../types';
+import { Viewport } from '@iot-app-kit/core';
+
+// known bug with zooming into live data https://github.com/apache/echarts/issues/11679
+// this function will prevent the data from moving if we are zoomed in
+const onDataZoomEvent = (chart: EChartsType, setViewport: (viewport: Viewport, lastUpdatedBy?: string) => void) => {
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  const currentDataZoom = chart.getOption().dataZoom[0];
+
+  const newDataZoom = {};
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  currentDataZoom.start > 0 ? (newDataZoom.startValue = currentDataZoom.startValue) : (newDataZoom.start = 0);
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  //@ts-ignore
+  currentDataZoom.end < 100 ? (newDataZoom.endValue = currentDataZoom.endValue) : (newDataZoom.end = 100);
+
+  setViewport(
+    { start: new Date(currentDataZoom.startValue), end: new Date(currentDataZoom.endValue) },
+    ECHARTS_GESTURE
+  );
+  chart.setOption(
```

Comment:
do we need to set this instantly ? can this hook return this object so all the options is set in the set options hook

Reply:
yes this needs to be done instantly as it is a user scroll event which we are overriding

Reply:
hmmmmm, mind taking at the useTrendCursors hook ??

Reply:
I am very positive that this need not be called here ....

Reply:
https://github.com/awslabs/iot-app-kit/blob/main/packages/react-components/src/components/chart/baseChart.tsx#L40